// Copyright (c) 2012 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.
//
// Protocol buffer definitions for Syncable FileSystem.

syntax = "proto2";

option optimize_for = LITE_RUNTIME;

package sync_file_system;

// TODO(tzik): Migrate from DriveMetadat to DriveEntryMetadata and erase this.
// http://crbug.com/240165
message DriveMetadata {
  required string resource_id = 1;
  required string md5_checksum = 2;
  required bool conflicted = 3;

  // When true, indicates that the remote file identified by |resource_id|
  // should be fetched for solving the conflict (resolve-to-remote resolution).
  required bool to_be_fetched = 4;

  enum ResourceType {
    RESOURCE_TYPE_FILE = 0;
    RESOURCE_TYPE_FOLDER = 1;
  }

  optional ResourceType type = 5 [default = RESOURCE_TYPE_FILE];
}
